core/templates/pages/exploration-player-page/layout-directives/audio-bar.directive.ts

```diff
@@ -77,10 +77,12 @@ angular.module('oppia').directive('audioBar', [
           };
 
           $scope.expandAudioBar = function() {
+            $scope.$emit('audioBarIsExpanded');
```

Comment:
Please don't use $broadcast, $emit or $on. Use a service if you need to maintain state, or pass arguments through the directive.

Reply:
Created a service.